### PR TITLE
feat: #92 add notice component without styles and storyBook

### PR DESCRIPTION
**Pull request by anterebol**
The StoryBook is in process...

### CODE REVIEW

File: mw-webapp/src/component/notice/Notice.module.scss
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.module.scss:1]: If you don't use styles for component then delete this file

File: mw-webapp/src/component/notice/Notice.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import {useEffect} from "react";
+import {Button} from "../button/Button";
+import {Title} from "../title/Title";
+import {HeadingLevel} from "../title/Title";
+import styles from "./Notice.module.scss";
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:5]: WE use absolute paths everywhere

Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:5]: We use absolute paths everywhere

File: mw-webapp/src/component/notice/Notice.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+const MILLISECONDS = 1000;
+
+export interface NoticeProps {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:9]: Add js-doc to each export public members

File: mw-webapp/src/component/notice/Notice.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import {useEffect} from "react";
+import {Button} from "../button/Button";
+import {Title} from "../title/Title";
+import {HeadingLevel} from "../title/Title";
+import styles from "./Notice.module.scss";
+
+const MILLISECONDS = 1000;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:7]: We no need this const in components. Id you will need it you can get it by props from logic layer. Am I wrong?

File: mw-webapp/src/component/notice/Notice.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+   */
+  description: string;
+  /**
+   * Type for notice, like 'Error', 'Ok', 'Warning' (text)
+   */
+  type: string;
+  /**
+   * Timer for automaticle closing (seconds)
+   */
+  timer: number;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:29]: Do we always need this property? Maybe better make it optional. For example I need to read notification few times? then better don't close it automatically

File: mw-webapp/src/component/notice/Notice.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  /**
+   * Callback which close the notification
+   */
+  closeNotice: (id: string) => void;
+}
+
+export const Notice = ({id, title, description, type, timer, closeNotice}: NoticeProps) => {
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:36]: 1/ add js-doc to public member
2/ don't use destructure in props. Image that you have 100 props/ It's will hard to write everything

File: mw-webapp/src/component/notice/Notice.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        level={HeadingLevel.h3}
+        className={styles.title}
+        text={title}
+      />
+      <p className={[styles.description].join(" ")}>
+        {description}
+      </p>
+      <Button
+        value={"Ok"}
+        onClick={() => closeNotice(id)}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:58]: Can't we write onClick={closeNotice(id)} ?

File: mw-webapp/src/component/notice/Notice.tsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+      <Title
+        level={HeadingLevel.h3}
+        className={styles.title}
+        text={title}
+      />
+      <p className={[styles.description].join(" ")}>
+        {description}
+      </p>
+      <Button
+        value={"Ok"}
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:57]: If value === "OK" everywhere then make it const if value can be difference - need get it from props

File: mw-webapp/src/component/notice/Notice.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    <article
+      className={[styles.notice, styles[type]].join(" ")}
+      id={id}
+    >
+      <Title
+        level={HeadingLevel.h3}
+        className={styles.title}
+        text={title}
+      />
+      <p className={[styles.description].join(" ")}>
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:53]: Can you please explain what this class means? Why you need join?

File: mw-webapp/src/component/notice/Notice.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+export const Notice = ({id, title, description, type, timer, closeNotice}: NoticeProps) => {
+
+  const millisecondsTimer = timer * MILLISECONDS;
+
+  useEffect(() => {
+    setTimeout(() => closeNotice(id), millisecondsTimer);
+  }, []);
+  return (
+    <article
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/notice/Notice.tsx:44]: Why tag article?

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/notice/Notice.tsx (hunk 5 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:36]: Ok, will change it.

File: mw-webapp/src/component/notice/Notice.tsx (hunk 4 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:29]: Sure. I'm agree with you.

File: mw-webapp/src/component/notice/Notice.tsx (hunk 9 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:44]: I thougt to write semantic tag. Do you prepare usual "div"

File: mw-webapp/src/component/notice/Notice.tsx (hunk 8 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:53]: Sorry, it's my mistake, forgot to delete.

File: mw-webapp/src/component/notice/Notice.tsx (hunk 3 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:7]: We receive timer (seconds) inside the props, but setTimeout use miliseconds inside. We just convert it.

File: mw-webapp/src/component/notice/Notice.tsx (hunk 7 above)
Comment (anterebol) [reply mw-webapp/src/component/notice/Notice.tsx:57]: I would suggest using the cross button in the angle, but for it I think we need to create another component.

File: mw-webapp/src/component/notice/Notice.tsx (hunk 9 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/notice/Notice.tsx:44]: No, no div. Just ask is it the best semabtic tag for notifications

**Pull request closed without merging**